libs/sdk-ui-kit/src/Dialog/ShareDialog/test/ShareDialog.test.tsx

```diff
@@ -0,0 +1,244 @@
+            ["add not group all", "shared", false],
+        ])(
+            "should %s when status of sharedObject is%s",
+            async (_desc: string, status: ShareStatus, visible: boolean) => {
+                const sharedObject: ISharedObject = { ...defaultSharedObject, shareStatus: status };
+                const wrapper = await createComponent({ sharedObject });
+                expect(isGroupAllVisible(wrapper)).toBe(visible);
+            },
+        );
+
+        it.each([
+            ["inactive owner", "is not specified", undefined],
+            ["active owner", "is specified", workspaceUser],
+        ])("should render %s when createdBy user %s", async (_desc: string, _desc2: string, user: IUser) => {
+            const sharedObject: ISharedObject = { ...defaultSharedObject, createdBy: user };
+            const wrapper = await createComponent({ sharedObject });
+            const isUserActive = user !== undefined;
+
+            expect(isOwnerVisible(wrapper, isUserActive)).toBe(true);
+        });
+
+        it("should mark current user in grantees ", async () => {
+            const userRef = userAccessGrantee.user.ref;
+            const wrapper = await createComponent({ currentUserRef: userRef }, [], [], [userAccessGrantee]);
+            expect(isCurrentUserInGrantees(wrapper)).toBe(true);
+        });
+
+        it("should mark current in grantees when is owner", async () => {
+            const sharedObject: ISharedObject = { ...defaultSharedObject, createdBy: workspaceUser };
+            const wrapper = await createComponent({ sharedObject, currentUserRef: workspaceUser.ref });
+            expect(isCurrentUserInGrantees(wrapper)).toBe(true);
+        });
+
+        it("should remove grantee from list and submit callback for empty grantees list status is private", async () => {
+            const granteeSelector = getGranteeSelector(userAccessGrantee.user);
+            const onApply = jest.fn();
+            const expectedPayload: ISharingApplyPayload = {
+                granteesToAdd: [],
+                granteesToDelete: [
+                    {
+                        granteeRef: userAccessGrantee.user.ref,
+                    },
+                ],
+                isUnderStrictControl: false,
+                isLocked: false,
+                shareStatus: "private",
+            };
+
+            const wrapper = await createComponent({ onApply: onApply }, [], [], [userAccessGrantee]);
+            clickDeleteGranteeIcon(wrapper, granteeSelector);
+
+            expect(isGranteeVisible(wrapper, granteeSelector)).toBe(false);
+
+            shareDialogSubmit(wrapper);
+
+            expect(onApply).toHaveBeenCalledTimes(1);
+            expect(onApply).toHaveBeenLastCalledWith(expectedPayload);
+        });
+
+        it("should remove grantee from list and submit callback for empty grantees list status is private and isUnderStrictControl is true ", async () => {
```

Comment:
"grantees list when object is private and isUnderStrictControl is true"

Reply:
All test removed it is duplicate